src/FileSystem.cpp

```diff
@@ -0,0 +1,129 @@
+#include "FileSystem.h"
+std::shared_ptr<FSDir> FileSystem::root=nullptr;
+FileSystem::FileSystem(int argc, char ** argv){
+	
+	FileSystem::root=std::make_unique<FSDir>("/");
+	/*DEBUG directories
+	((FSDir*)getFileHandle("/").get())->createDir("BDMV");
+	((FSDir*)getFileHandle("/").get())->createDir("AACS");
+	((FSDir*)getFileHandle("/").get())->createFile("test","ala ma kota");
+	*/
+	//Set up C library callbacks...
+	static struct fuse_operations operations{};
+		operations.getattr = do_getattr;
+		operations.readdir = do_readdir;
+		operations.read    = do_read;
+		operations.create  = do_create;
+		operations.mkdir   = do_mkdir;
+		operations.utimens = do_times;
+		operations.write   = do_write;
```

Comment:
One more operation that would be usef is `do_open` - it doesn't have to do anything but currently it is not possible to write to an existing file...

```
$ echo TEST > f1      
$ echo TEST > f1
zsh: function not implemented: f1
```